std-bits/table/src/main/java/org/enso/table/data/column/storage/BoolStorage.java

```diff
@@ -391,4 +398,45 @@ public BoolStorage slice(List<SliceRange> ranges) {
 
     return new BoolStorage(newValues, newMissing, newSize, negated);
   }
+
+  @Override
+  public Storage<?> cast(StorageType targetType, CastProblemBuilder castProblemBuilder) {
+    return switch (targetType) {
+      case AnyObjectType any ->
+        new MixedStorageFacade(this);
+      case BooleanType booleanType ->
+        this;
+      case FloatType floatType -> {
+        int n = size();
+        long[] newData = new long[n];
+        long t = Double.doubleToRawLongBits(1.0);
+        long f = Double.doubleToRawLongBits(0.0);
+        for (int i = 0; i < n; i++) {
+          newData[i] = values.get(i) ? t : f;
+        }
```

Comment:
this feels like leaking the internal implementation of the `DoubleStorage`.

Reply:
Fair, will fix